lib/mfg.c

```diff
@@ -163,8 +183,16 @@ static int secure_config_get(struct switchtec_dev *dev,
 				*otp_valid = true;
 		}
 	} else {
-		ret = switchtec_mfg_cmd(dev, MRPC_SECURITY_CONFIG_GET, NULL, 0,
-					&reply, sizeof(reply));
+		if (switchtec_gen(dev) == SWITCHTEC_GEN4) {
```

Comment:
It would be better if the I2c changes and MRPC changes were in different patches...